systems/matrix.asm

```diff
@@ -0,0 +1,15 @@
+section .text
+global index
+index:
+	; rdi: matrix
+	; rsi: rows
+	; rdx: cols
+	; rcx: rindex
+	; r8: cindex
+
+	mov rax, rdx ; move cols to rax (to calculate row offset)
+	mul rcx ; get row offset in rax
+	add rax, r8 ; add col offset to rax
+	imul rax, 4 ; convert offset units from index to bytes
```

Comment:
This could be incorporated into the `mov` instruction. I'm not sure if that actually saves any micro-ops, but at the very least you get a shorter program :)